ultravox/data/datasets.py

```diff
@@ -1180,9 +1186,9 @@ def __iter__(self):
                 iters[iter_index] = iter(self._datasets[iter_index])
                 yield next(iters[iter_index])
 
-    def __len__(self):
+    def __len__(self) -> int:
         # TODO: Implement the length method for different stop strategies
-        return sum(len(ds) for ds in self._datasets)
+        return sum(int(getattr(ds, "weight", 1) * len(ds)) for ds in self._datasets)
```

Comment:
Shouldn't `len` be conditional here?
IMO the concept of len doesn't make much sense when `self._using_epochs` is False.